### PR TITLE
:bug: Fixes flakey tests

### PR DESCRIPTION
**Pull request by gowthamidommety-okta**
Fixes a windows hello test.

### CODE REVIEW

**Comment by gowthamidommety-okta** [comment]
@mauriciocastillosilva-okta @rchild-okta  Please review

File: test/unit/spec/MfaVerify_spec.js [hunk 1]
```diff
@@ -174,6 +174,39 @@ function (Okta,
       });
     }
 
+    function setupWebauthnOnly() {
```
Comment (rchild-okta) [inline test/unit/spec/MfaVerify_spec.js]: Can you explain the reason why we need to break this out into a separate function (which looks very similar to the other setup function)?

Comment (gowthamidommety-okta) [inline test/unit/spec/MfaVerify_spec.js]: The other setup function has `Expect.waitForMfaVerify()`, and for this particular test, where Windows Hello is the only factor, it never goes to that controller.

File: test/unit/spec/MfaVerify_spec.js [hunk 2]
```diff
@@ -174,6 +174,39 @@ function (Okta,
...
+      var setNextResponse = Util.mockAjax();
+      var baseUrl = 'https://foo.com';
+      var authClient = new OktaAuth({url: baseUrl, transformErrorXHR: LoginUtil.transformErrorXHR});
+      var successSpy = jasmine.createSpy('success');
+      var router = new Router(_.extend({
```
Comment (mauriciocastillosilva-okta) [inline test/unit/spec/MfaVerify_spec.js]: can we refactor part of this function into one that returns the router? And that way less code is duplicated

File: test/unit/spec/MfaVerify_spec.js [hunk 3]
```diff
@@ -2002,9 +2033,15 @@ function (Okta,
           return emulateWindows()
           .then(setupWebauthnOnly)
           .then(function (test) {
+            return Expect.waitForVerifyWindowsHello(test);
```
Comment (mauriciocastillosilva-okta) [inline test/unit/spec/MfaVerify_spec.js]: can we move this ```Expect.waitForVerifyWindowsHello``` into ```setupWebauthnOnly````?

File: test/unit/spec/MfaVerify_spec.js [hunk 4]
```diff
@@ -174,6 +174,39 @@ function (Okta,
...
+      Util.registerRouter(router);
+      Util.mockRouterNavigate(router);
+      setNextResponse([resRequiredWebauthn, resChallengeWebauthn, resSuccess]);
+      router.refreshAuthState('dummy-token');
+      var $forms = $sandbox.find('.o-form');
+      var forms = _.map($forms, function (form) {
+        return new MfaVerifyForm($(form));
+      });
+      if (forms.length === 1) {
+        forms = forms[0];
```
Comment (mauriciocastillosilva-okta) [inline test/unit/spec/MfaVerify_spec.js]: can you tell me why we need this? This setup function is only used in one test, so we probably know if this form is present or not.

File: test/unit/spec/MfaVerify_spec.js (hunk 2 above)
Comment (gowthamidommety-okta) [inline test/unit/spec/MfaVerify_spec.js]: I think creating another function here may cause a lint error similar to my first commit.

`/home/travis/build/okta/okta-signin-widget/test/unit/spec/MfaVerify_spec.js
  114:33  error  Function has too many statements (35). Maximum allowed is 34  max-statements`
 
Should I create another describe? Is there a better way around it?

File: test/unit/spec/MfaVerify_spec.js (hunk 4 above)
Comment (gowthamidommety-okta) [inline test/unit/spec/MfaVerify_spec.js]: Looking at this, I actually think we don't need any of it. Our test only references router, so returning a reference to that should be enough.

File: test/unit/spec/MfaVerify_spec.js (hunk 3 above)
Comment (gowthamidommety-okta) [inline test/unit/spec/MfaVerify_spec.js]: Will change

File: test/unit/spec/MfaVerify_spec.js (hunk 2 above)
Comment (mauriciocastillosilva-okta) [inline test/unit/spec/MfaVerify_spec.js]: you can change, at the top of the file, the allowed max-statements, and set it to 35, or whatever number you need

Comment (gowthamidommety-okta) [inline test/unit/spec/MfaVerify_spec.js]: 👍 done

**Pull request merged**